### PR TITLE
Fixed Require MRN patient setting - see issue #70

### PR DESCRIPTION
**Pull request by mikejmets**
## Description of the issue/feature this PR addresses

Linked issue: https://github.com/senaite/senaite.patient/issues/70

## Desired behavior after PR is merged

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by mikejmets** [comment]
I set this to Draft because the tests I created fail. Namely, when the Require MRN setting is True and I create a Patient with no MRN value, the patient is created. My test expects it to fail. Am I wrong?

File: src/senaite/patient/tests/doctests/Patient.rst [hunk 1]
```diff
@@ -46,3 +48,23 @@ A patient can have a primary and additional email addresses:
...
+    True
+    >>> patient = api.create(patients, "Patient")
+    >>> patient
+    None
+    >>> plone_api.portal.set_registry_record("senaite.patient.require_patient", False)
+    >>> patient_api.is_patient_required()
+    False
+    >>> patient = api.create(patients, "Patient")
+    >>> patient
+    <Patient at /plone/patients/P000002>
```
Comment (ramonski) [inline src/senaite/patient/tests/doctests/Patient.rst:63]: The validator is not called when the field is set. It is only called when the form is submitted, e.g. like `processForm` in AT but it is called `applyValue` here:
https://github.com/zopefoundation/z3c.form/blob/master/src/z3c/form/widget.py#L404

Therefore, this test will always pass unless you do the same validation dance inside the custom filed setter.

**Review by ramonski — CHANGES_REQUESTED** [review]

File: src/senaite/patient/adapters/listing.py [hunk 2]
```diff
@@ -120,7 +120,10 @@ def folder_item(self, obj, item, index):
         item["PatientID"] = sample_patient_id
 
         # get the patient object
-        patient = self.get_patient_by_mrn(sample_patient_mrn)
+        if patient_api.is_patient_required():
+            patient = self.get_patient_by_mrn(sample_patient_mrn)
+        else:
+            patient = self.get_patient_by_id(sample_patient_id)
```
[the remark this answers is not in the document]
Comment (mikejmets) [reply src/senaite/patient/adapters/listing.py]: Done

**Comment by ramonski** [comment]
Thanks for the fix Mike. Would you mind to add a Changelog entry, then we can merge it.

**Comment by mikejmets** [comment]
Changelog updated

**Review by ramonski — APPROVED** [review]

**Pull request merged**